### PR TITLE
Autonomous chooser

### PR DESCRIPTION
**Pull request by BrownGenius**

### CODE REVIEW

File: src/main/java/bhs/devilbotz/RobotContainer.java [hunk 1]
```diff
@@ -78,10 +78,35 @@ private void buildAutoCommands() {
...
+        case MOBILITY_DOCK_AND_ENGAGE:
+          break;
+
+        case SCORE_DOCK_AND_ENGAGE:
+          break;
+
+        case SCORE_MOBILITY_DOCK_ENGAGE:
+          break;
+        case SCORE_MOBILITY_PICK_DOCK_ENGAGE:
+          break;
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/RobotContainer.java:106]: you shouldn't need any of this, its a redundant check The autoCommands.get will do all of this for you.

File: src/main/java/bhs/devilbotz/lib/AutonomousModes.java [hunk 2]
```diff
@@ -6,6 +6,15 @@
...
  */
 public enum AutonomousModes {
+  SIT_STILL,
+  MOBILITY,
+  SCORE_AND_MOBILITY,
+  DOCK_AND_ENGAGE,
+  MOBILITY_DOCK_AND_ENGAGE,
+  SCORE_DOCK_AND_ENGAGE,
+  SCORE_MOBILITY_DOCK_ENGAGE,
+  SCORE_MOBILITY_PICK_DOCK_ENGAGE,
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/lib/AutonomousModes.java:16]: Write javadocs please

File: src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java [hunk 3]
```diff
@@ -46,24 +57,53 @@ public ShuffleboardManager() {
...
-        "Drive Distance Straight PID", AutonomousModes.DRIVE_STRAIGHT_DISTANCE_PID);
+    autoMode.add("Routine", autoModeChooser).withWidget(BuiltInWidgets.kComboBoxChooser);
+  }
+
+  private void initAutoModePreferences() {
+    autoDelay =
+        autoMode.add("Delay (in seconds)", 0).withWidget(BuiltInWidgets.kTextView).getEntry();
+
+    autoDistance =
+        autoMode.add("Distance (in meters)", 5).withWidget(BuiltInWidgets.kTextView).getEntry();
```
Comment (ParkerMeyers) [inline src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java:93]: Do these work or are they just theoretical for the future?

**Review by ParkerMeyers — CHANGES_REQUESTED** [review]
Just fix the redundant check with the autonomous mode

File: src/main/java/bhs/devilbotz/RobotContainer.java (hunk 1 above)
Comment (ParkerMeyers) [reply src/main/java/bhs/devilbotz/RobotContainer.java:106]: (your default case)

Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/RobotContainer.java:106]: Hi Parker, the code isn't redundant.  `buildAutoCommands` is initialized in the constructor so those commands can't use any of the user defined values from Shuffleboard. `getAutonomousCommand` is called when the autonomous routine is just about to be executed, so the command can be created on the fly using the Shuffleboard parameters.  So, eventually, other than the "Sit Still" command, the autonomous commands need to be dynamically created.  All of the autonomous routines defined [here](https://github.com/BHSRobotix/ChargedUp2023#desired-autonomous-routines) have at least one configurable parameter.

File: src/main/java/bhs/devilbotz/lib/AutonomousModes.java (hunk 2 above)
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/lib/AutonomousModes.java:16]: Yes, will do!

File: src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java (hunk 3 above)
Comment (BrownGenius) [reply src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java:93]: autoDistance is currently used with the "Mobility" command.  E.g. you can specify the distance, select "Mobility" in the autonomous chooser and then enable autonomous mode.  Using @joshuman's `DriveStraughtPID` class, it should travel the distance requested in Shuffleboard. You can play with it in simulation.

**Review by ParkerMeyers — APPROVED** [review]

**Comment by BrownGenius** [comment]
Resolves #65

**Pull request merged**